service/arm_templates/node_setup.json

```diff
@@ -179,22 +260,27 @@
             "version": "[parameters('osImageVersion')]"
           },
           "osDisk": {
-            "name": "[variables('osDiskName')]",
-            "vhd": {
-              "uri": "[concat('http://', parameters('storageAccountName'), '.blob.core.windows.net/vhds/', variables('osDiskName'), '.vhd')]"
-            },
+            "name": "[concat(parameters('clusterID'), '-', parameters('prefix'), '-OSDisk')]",
             "caching": "ReadWrite",
+            "managedDisk": {
+              "storageAccountType": "[variables('storageAccountType')]"
+            },
             "createOption": "FromImage"
           },
-          "dataDisks": [
+          "copy": [
             {
-              "name": "[variables('dataDiskName')]",
-              "diskSizeGB": "[parameters('dataDiskSizeGB')]",
-              "lun": 0,
-              "vhd": {
-                "uri": "[concat('http://', parameters('storageAccountName'), '.blob.core.windows.net/vhds/', variables('dataDiskName'), '.vhd')]"
-              },
-              "createOption": "Empty"
+              "name": "dataDisks",
+              "count": "[length(parameters('dataDisks'))]",
+              "input": {
+                "name": "[concat(parameters('clusterID'), '-', parameters('prefix'), '-', parameters('dataDisks')[copyIndex('dataDisks')].name)]",
+                "caching": "ReadWrite",
+                "diskSizeGB": "[parameters('dataDisks')[copyIndex('dataDisks')].sizeGB]",
+                "managedDisk": {
+                  "storageAccountType": "[variables('storageAccountType')]"
+                },
+                "createOption": "Empty",
+                "lun": "[copyIndex('dataDisks')]"
+              }
```

Comment:
I think now we should add some mount service extensions to the cloudconfig, right?

Reply:
right, but let's do it later, when disks will be in place.